File: project0.cpp
//////////////////////////////////////////////////////// PROJECT 0 //////////////////////////////////////////////////////////////////////////////////////////


#include <iostream>
#include <fstream>
#include <string.h>

using namespace std;

ifstream input_file_0; // declare a pointer to an input file 
ofstream out_f("output.txt",ios::out); // declare the output file

class P0_COURSE{
   public:      // public methods for this class

		P0_COURSE(string, int); // example: b.P0_COURSE("EE59", 4);
			// constructor;
			// populates the private data members from the 
			// data read from the input file called studentInfo.txt;

		void P0_LIST(int);  // example: b.P0_LIST(x);
			// if x is 0, give the course information;
			// if x is 1, give more information;
			// if x has any other value, input error;
			// returns no values;

	protected: // private var to be used by this class and allows for
			// other classes inherit from this class;

		int n;  // noof students
		string courseName; // string to keep the course name;
		int id[20]; // the integer array to hold the student ids
		int pin[20]; // the integer array to hold the pin numbers
};

// start constructor code:
P0_COURSE::P0_COURSE(string x, int y)
{
	int i;
	// x is the course name
	courseName = x;
	// y is number of students
	n = y;

	input_file_0.open("studentInfo.txt", ios::in); // open input file;

        for(i=0; i<n; i++)
        {
			input_file_0 >> id[i] >> pin[i];
        }

	input_file_0.close(); // close input file;
	input_file_0.clear(); // rewind file pointer to the top of the file;

	// debugging print statement: prints on the terminal screen; 
	for(i = 0; i < n; i++) 
	{ 
		cout << "IN P0: id[" << i << "] " << id[i] << " pin[" << i << "] " << pin[i] << endl; 
	}
	out_f << "++ P0 ++ START ++++" << endl;
	out_f << "++ P0 CONSTRUCTOR SUCCESSFULLY INSTANTIATED A P0_COURSE OBJECT." << endl;
	out_f << "++ P0 NAME OF THE COURSE IS " << courseName << "." << endl;
	out_f << "++ P0 THERE ARE " << n << " STUDENTS IN THIS P0_COURSE OBJECT." << endl;
	out_f << "++ P0 ++ END  ++++" << endl;
} // end of constructor

// start P0_LIST code:
void P0_COURSE::P0_LIST(int x)
{ 
	// your code goes here:
	int i;
	if (x==0)
	{
		out_f << "++ P0 ++ START OUTPUT FROM P0_LIST METHOD:" << endl;
		out_f << "++ P0 THE COURSE IS " << courseName << " WITH " << n << " STUDENTS." << endl;
		out_f << "++ P0 ++ END OUTPUT FROM P0_LIST METHOD." << endl;
	}	

	else if (x==1)
	{
		out_f << "++ P0 ++ START OUTPUT FROM P0_LIST METHOD:" << endl;
		out_f << "++ P0 THE COURSE IS " << courseName << " WITH " << n << " STUDENTS." << endl;
		for(i = 0; i < n; i++) 
		{
			out_f << "++ P0 " << id[i] << endl;
		}
		out_f << "++ P0 ++ END OUTPUT FROM P0_LIST METHOD." << endl;	
	}
	
	else 
	{
		out_f << "++ P0 ++ START OUTPUT FROM P0_LIST METHOD:" << endl;
		out_f << "++ P0 SINCE P0_LIST IS CALLED WITH " << x << ", INPUT ERROR." << endl;
		out_f << "++ P0 ++ END OUTPUT FROM P0_LIST METHOD." << endl;

	}
} // end of P0_LIST code

File: Projects/Project0/python/project0.py
//////////////////////////////////////////////////// PROJECT 0 python //////////////////////////////////////////////////////////////////////////////////////////


output_file = open("python_Output.txt",'w')
class P0_COURSE:

	def __init__(self,course_name, x):
		self.courseName = course_name # name of the course
		self.n = x # number of customers
		print(f"x = {self.n}\n")
		print(f"courseName = {self.courseName}\n")

        # creating a Python array of a specific length and initializing all elements to 0 
		# Note that these arrays are not static, we can append to or pop from these lists:
		self.id = [0 for _ in range(x)] # same as int id[x] = {0, 0, 0, 0, 0, ... etc.} in C++
		self.pin = [0 for _ in range(x)] # same as int pin[x] = {0, 0, 0, 0, 0, ... etc.} in C++

        # not as conventional way in Python but closer to the original sample_p0.h 
		i = 0 
		with open('studentInfo.txt','r') as i_f:
			for line in i_f: 
				self.id[i] = line.split()[0] # read first word from the line and put it into id array
				self.pin[i] = line.split()[1] # read second word from the line and put it in pin array
				i += 1
				print(i)

		output_file.write (f"++ P0 ++ START ++++\n")
		output_file.write (f"++ P0  CONSTRUCTOR SUCCESSFULLY INSTANTIATED A P0_COURSE OBJECT.\n");
		output_file.write (f"++ P0  NAME OF THE COURSE IS {self.courseName}.\n")
		output_file.write (f"++ P0  THERE ARE {self.n} STUDENTS IN THIS P0_COURSE OBJECT.\n")
		output_file.write (f"++ P0 ++ END ++++ \n")
    
	def P0_LIST(self, x):
		print(f"x is {x}\n")
		output_file.write ("++ P0 ++ START OUTPUT FROM P0_LIST METHOD:\n")
		if (x == 0):
			output_file.write (f"++ P0 THE COURSE IS {self.courseName} WITH {self.n} STUDENTS.\n")
			output_file.write ("++ P0 ++ END OUTPUT FROM P0_LIST METHOD.\n")
		elif (x == 1):
			output_file.write (f"++ P0 THE COURSE IS {self.courseName} WITH {self.n} STUDENTS.\n")
			print(f"x is {x}\n")
			for i in range (self.n):
				output_file.write(f"++ P0 {self.id[i]}\n")
		else:
			print(f"x is {x}\n")
			output_file.write(f"++ P0 SINCE P0_LIST IS CALLED WITH {x}, INPUT ERROR.\n")
		output_file.write("++ P0 ++ END OUTPUT FROM P0_LIST METHOD.\n")
